scripts/go_release.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Helper script to prepare a Go Release for the SDKs.
+
+set -e
+
+# Function to validate the version number format x.y.z(-beta.w)
+validate_version_number() {
+    local version="$1"
+    if [[ "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-beta\.[0-9]+)?$ ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Function to validate the build number format Mmmppbb
+validate_build_number() {
+    local build="$1"
+    if [[ "$build" =~ ^[0-9]{1}[0-9]{2}[0-9]{2}[0-9]{2}$ ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+# Read and validate the version number
+while true; do
+    read -p "Please provide the version number (x.y.z(-beta.w)): " version_number
+    if validate_version_number "$version_number"; then
+      break
+   else
+      echo "Invalid version number format. Please try again."
+    fi
+done
+
+# Read and validate the build number
+while true; do
+    read -p "Please provide the build number (Mmmppbb): " build_number
+    if  validate_build_number "$build_number"; then
+      break    
+   else
+      echo "Invalid build number format. Please try again."
+   fi
+done
+
+# Replace version number with new version number
+awk -v build="$build_number" -v version="$version_number" '
+  /SDKSemverVersion/ && !done {
+    print "   SDKSemverVersion      =", "\"" build "\" // v" version;
+    done = 1;
+    next;
+  } 
+  { print }' internal/core.go > tmpfile && mv tmpfile internal/core.go
```

Comment:
This should not be necessary.

We could simply use a `version` file, and `go:embed` this into the `SDKSemverVersion` variable. Let me know if you'd like to pair up on this.

Reply:
Done!